app/assets/javascripts/admin/controllers/tree_app.coffee

```diff
@@ -1,51 +1,34 @@
-#admin
-#admin
-#admin
-#admin
-#admin
-#admin
-#admin
-#admin
-
-
-
 app = angular.module('treeApp')
 
 app.controller 'BodyController', ($scope, Order, OrderItem) ->
   $scope.cart = Order.cart()
 
   #  This add_product uses item.id for the product page.
   $scope.add_product = (item) ->
+    if item.quantity == 0
+      $scope.remove = (item) ->
+        oi = new OrderItem(source_id: item.id, source_type: "Product", quantity: item.quantity, order_id: $scope.cart.id)
+        oi.$save()
```

Comment:
I moved the oi out of the if, and simplified the if block to just either deleting or saving.  This just checked to see if the item was already in the cart, then if its quantity was changed to zero, delete the item from the cart.